### ISSUE TITLE
State tax reforms produce no impact in microsimulation

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/policyengine-us#4538]
For example, increasing the California Young Child Tax Credit to $2,000 has no impact in the [web](https://policyengine.org/us/policy?focus=policyOutput.policyBreakdown&reform=55047&region=ca&timePeriod=2024&baseline=2) or [Python](https://colab.research.google.com/drive/1-BNJGAngkhGFWSJ-YU7BXq4N0R0sVVyZ?usp=sharing) interfaces. Same for [zeroing out the Washington Working Families Tax Credit](https://policyengine.org/us/policy?reform=55008&focus=policyOutput.policyBreakdown&region=wa&timePeriod=2024&baseline=2&household=44418), despite it working [at the household level](https://policyengine.org/us/household?reform=55008&focus=householdOutput.earnings&region=wa&timePeriod=2024&baseline=2&household=44418).

#4488 looked to add the `exhaustive_parameter_dependencies` correctly but I'd guess it's broken.

**Comment by anth-volk** [comment]
Did some testing on this. Running the notebook that you provided against different `policyengine_us` versions, it's actually caused by the updates that came with v0.767.0 (https://github.com/PolicyEngine/policyengine-us/compare/0.766.0...0.767.0), which incorporated CBO income elasticities and self-employment income responses.

**Comment by anth-volk** [comment]
Okay, I think I found the issue. If you reference the changes that occurred with v0.767.0, in `policyengine_us/system.py`, the block that backdates parameters now occurs before the state variables are initialized instead of after. In testing, when I moved the parameter backdating block after the below code block and tested, I received the same calculated output for `gov.states.ca.tax.income.credits.young_child.amount` as prior to v0.767.0.

```
        for parameter in self.parameters.get_descendants():
            parameter.modified = False

        if reform is not None:
            reform.apply(self)

        self.add_variables(*create_50_state_variables())
```
